### ISSUE TITLE
Living armor extra health emptied on save/reload.

### ISSUE THREAD

**Issue by MentalMouse** [WayofTime/BloodMagic#1447]
#### Issue Description:
When the game is loaded in SSP, the player's current health is clipped to the normal max of 20.

#### What happens:
Living Armor's Healthy upgrade provides extra hearts.  When a single-player game is saved and reloaded, the extra hearts will be empty regardless of their prior state.  They will immediately start to heal if the food level is high enough.

#### What you expected to happen:
The player's current health should be saved and restored with the rest of the game state.

#### Steps to reproduce:

1. Set up player in SSP with Living Armor and the Healthy upgrade.
2. Save the game.
3. Reload the game.  Note the health bar.
...
____
#### Affected Versions (Do *not* use "latest"):

- BloodMagic: 2.3.3-CUSTOM
- Minecraft: 1.12.2
- Forge: 2705

**Comment by TehNut** [comment]
Yes, this is how vanilla's extra hearts function.

**Comment by MentalMouse** [comment]
I beg your pardon, but:
1. Failing to preserve game state over a save/restore is a **big deal**.  If vanilla did that, I'm pretty sure it would have been in the wiki (probably in bold) from ages ago, and it's not.
2. But of course the wiki is yadda yadda.  So, I just fired up a vanilla 1.12.2 game to test the point directly:
2a:  I used commands to give myself effect 21 (Health Boost).  
2b: As documented, the new hearts appeared empty, so I dipped into creative for a stack of steak, and finished "healing" them.  
2c: Save and Load... yep, still at full boosted health, as expected.  Vanilla maintains game state correctly.
